aspnetcore/fundamentals/portable-object-localization.md

```diff
@@ -0,0 +1,218 @@
+- `msgid`: The untranslated string.
+- `msgstr`: The translated string.
+
+In the case of pluralization support, more entries can be defined.
+
+- `msgid_plural`: The untranslated plural string.
+- `msgstr[0]`: The translated string for the case 0.
+- `msgstr[N]`: The translated string for the case N.
+
+The PO file specification can be found [here](https://www.gnu.org/savannah-checkouts/gnu/gettext/manual/html_node/PO-Files.html).
+
+## Configuring PO file support in ASP.NET Core
+
+This example is based on an ASP.NET Core MVC application generated from a Visual Studio 2017 project template.
+
+### Referencing the package
+
+Add a reference to the `OrchardCore.Localization.Core` NuGet package. It is available on [MyGet](https://www.myget.org/) at the following package source: https://www.myget.org/F/orchardcore-preview/api/v3/index.json
+
+The *.csproj* file now contains a line similar to the following (version number may vary):
+
+[!code-xml[Main](localization/sample/POLocalization/POLocalization.csproj?range=9)]
+
+### Registering the service
+
+Add the required services to the `ConfigureServices` method of *Startup.cs*:
+
+[!code-csharp[Main](localization/sample/POLocalization/Startup.cs?name=snippet_ConfigureServices&highlight=4-21)]
+
+Add the required middleware to the `Configure` method of *Startup.cs*:
+
+[!code-csharp[Main](localization/sample/POLocalization/Startup.cs?name=snippet_Configure&highlight=15)]
+
+Add the following code to your Razor view of choice. *About.cshtml* is used in this example.
+
+[!code-cshtml[Main](localization/sample/POLocalization/Views/Home/About.cshtml)]
+
+An `IViewLocalizer` instance is injected and used to translate the text `"Hello world!"`.
+
+### Creating a PO file
+
+Create a file named *fr.po*, in your application root folder, containing the following:
+
+[!code-text[Main](localization/sample/POLocalization/fr.po)]
+
+This file stores both the string to translate and the French-translated string. Translations revert to their parent culture, if necessary. In this example, the *fr.po* file is used if the requested culture is `fr-FR` or `fr-CA`.
+
+### Testing the application
+
+Run your application, and navigate to the URL `/Home/About`. The text **Hello world!** is displayed.
+
+Navigate to the URL `/Home/About?culture=fr-FR`. The text **Bonjour le monde!** is displayed.
+
+## Pluralization
+
+PO files support pluralization forms, which is useful when the same string needs to be translated differently based on a cardinality. This task is made complicated by the fact that each language defines custom rules to select which string to use based on the cardinality.
+
+The Orchard Localization package provides an API to invoke these different plural forms automatically.
+
+### Creating pluralization PO files
+
+Add the following content to the previously mentioned *fr.po* file:
+
+```text
+msgid "There is one item."
+msgid_plural "There are {0} items."
+msgstr[0] "Il y a un élément."
```

Comment:
Need to explain this and the next - that they correspond to sing/plural. Right now I need to guess that as I don't know french.